apps/web/src/lib/components/ui/agent-editor/create-agent.svelte

```diff
@@ -0,0 +1,156 @@
+<script lang="ts">
+	import { enhance } from '$app/forms';
+	import { Button, buttonVariants } from '$lib/components/ui/button';
+	import * as Dialog from '$lib/components/ui/dialog';
+	import { Input } from '$lib/components/ui/input';
+	import { Label } from '$lib/components/ui/label';
+	import { Textarea } from '$lib/components/ui/textarea';
+	import { Loader2 } from 'lucide-svelte';
+	import { createEventDispatcher } from 'svelte';
+	import { Toggle } from '$lib/components/ui/toggle';
+	import { invalidateAll } from '$app/navigation';
+	import { superForm } from 'sveltekit-superforms/client';
+
+	import { agentFormSchema, type AgentFormSchema } from '$lib/schema';
+	import { type SuperValidated } from 'sveltekit-superforms';
+	import { toast } from 'svelte-sonner';
+
+	export let data: SuperValidated<AgentFormSchema>;
+
+	const { form: formAgent, errors } = superForm(data, {
+		validators: agentFormSchema
+	});
+
+	export let form;
+
+	const dispatch = createEventDispatcher();
+
+	let state: 'loading' | 'error' | 'idle' = 'idle';
+
+	let open = false;
+	const handleTrigger = async () => {
+		open = true;
+	};
+
+	let published = false;
+</script>
+
+<Dialog.Root {open} onOpenChange={(o) => dispatch('close')}>
+	<div class="absolute bottom-5 right-5">
+	<Dialog.Trigger class={buttonVariants({ variant: 'outline' })} on:click={handleTrigger}
+		>
+			Create Agent
+			</Dialog.Trigger
+			>
+		</div>
+	<Dialog.Content class="w-full sm:max-w-full lg:max-w-4xl">
+		<Dialog.Header>
+			<Dialog.Title>Create Agent</Dialog.Title>
+		</Dialog.Header>
+		<form action="?/creatAgents" method="POST" use:enhance>
+			<div class="p-6">
+				<div class="flex w-full items-center gap-4">
+					<div class="mb-4 w-full">
+						<Label for="title" class="block text-sm font-medium ">Title</Label>
+						<Input
+							id="title"
+							name="title"
+							bind:value={$formAgent.title}
+							placeholder="Agent's title"
+							class="border-input placeholder:text-muted-foreground focus-visible:ring-ring col-span-3 mt-1  block  h-9 w-full  rounded-md border bg-transparent px-3 text-sm shadow-sm ring-offset-0 transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-offset-0 disabled:cursor-not-allowed disabled:opacity-50"
+						/>
+					</div>
+					<div class=" mb-4 mt-4 flex cursor-pointer items-center gap-4 p-2">
+						<Toggle
```

Comment:
I know I said "toggle" in the chat. but you should be well versed in the shadcn components by now to know that the "switch" component is probably better for this UI. correct me if I'm wrong though